README.md
# Material Alert Dialogs for Flutter 📱

A new Flutter package project.

## Getting Started

This project is a starting point for a Dart
[package](https://flutter.dev/developing-packages/),
a library module containing code that can be shared easily across
multiple Flutter or Dart projects.

For help getting started with Flutter, view our 
[online documentation](https://flutter.dev/docs), which offers tutorials, 
samples, guidance on mobile development, and a full API reference.

## Flutter package  to implement *animated*, 🥰 *Attractive*, 🎨*stylish* Material Dialog in Flutter easily.

<table style="width:100%">
  <tr>
    <th><b>1. Material Dialog<b></b></th>
    <th>2. Animated Material Dialog</th> 
  </tr>
  <tr>
    <td><img src="gifs/alert.gif"/></td>
    <td><img src="gifs/animated.gif"/></td>
  </tr>
</table>

# Introduction

**MaterialAlerDialog** library is built upon Flutter Material Design library. This API will be useful to create rich, animated, beautiful dialogs in Flutter  easily. 

<table style="width:100%">
  <tr>
    <th><b>1. Material Alert Dialog<b></b></th>
    <th>2. Animated Material Alert Dialog</th>
  </tr>
  <tr>
    <td>This is basic material dialog which has two material buttons (Same as Fluter's AlertDialog) as you can see below.</td>
    <td>This is Animated material alert  dialog which has two material buttons which is showed from bottom of device as you can see below.</td> 
  </tr>
  <tr>
    <td align="center"><img src="gifs/alert.jpg" width="75%"/></td>
    <td align="center"><img src="gifs/animated.jpg" width="75%"/></td> 
  </tr>
</table>

## Implementation
Implementation of Material Alert Dialog library is so easy. You can check [/app](/app) directory for demo. Let's have look on basic steps of implementation.

# Use this package as library
## 1.Depend on it
 Add this to your package's pubspec.yaml file:
 ```
 dependencies:
  material_alertdialog: ^0.0.1
 ```
## 2. Install it
You can install packages from the command line:
```
$ flutter pub get
```
Alternatively, your editor might support `flutter pub get.` Check the docs for your editor to learn more.

## 3. Import it
Now in your Dart code, you can use:
```
import 'package:material_alertdialog/material_alertdialog.dart';
```
### Create AlertDialog Instance
As there are two types of dialogs in library. Material Dialogs are instantiated as follows.
#### i. MaterialAlertDialog
`MaterialDialog` class is used to create MaterialAlertDialog. Its static `Builder` class is used to instantiate it. 
```java
       showDialog(context: context, builder:(context)=>
       MaterialAlertDialog(

                      height: 250,

                      title: "Exit",

                      message: "Are you sure you want to quit",

                     onPresedPositiveButton: (){
                       //add what you want for navigate
                     },

                      onPresedNegativeButton: (){
                        Navigator.pop(context);
                      },

                   ),
               );
```

<img align="center" src="gifs/alert.gif" width="300"/>


#### ii. AnimatedMaterialAlertDialog
`AnimatedMaterialAlertDialog` class is used to create MaterialAlertDialog. Its static `Builder` class is used to instantiate it. 
```java
       showDialog(context: context, builder:(context)=>
       AnimatedMaterialAlertDialog(
                      
                      imagePath: "assets/steth.jpg"
                      
                      height: 250,

                      title: "Exit",

                      message: "Are you sure you want to quit",

                     onPresedPositiveButton: (){
                     
                       //add what you want for navigate
                       
                     },

                      onPresedNegativeButton: (){
                        Navigator.pop(context);
                      },

                   ),
               );
```

<img style="text_align:center" align="center" src="gifs/animated.gif" width="300"/>

## Credits
This library is built using following open-source libraries.
- [Material Components for Flutter]
- [Material Dialog - Android - By Shreays Patil](https://github.com/PatilShreyas/MaterialDialog-Android/)



CHANGELOG.md
## [0.0.1] - TODO: Add release date.

* TODO: MaterialAlertDialog Library.


test/material_alertdialog_test.dart
import 'package:flutter/material.dart';
import 'package:material_alertdialog/animatedmaterial_alertdialog.dart';
import 'package:material_alertdialog/material_alertdialog.dart';


void main() => runApp(MyApp());

class MyApp extends StatelessWidget {
  // This widget is the root of your application.
  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      title: 'Flutter Demo',
      theme: ThemeData(
       
        primarySwatch: Colors.blue,
      ),
      home: MyHomePage(title: 'Material Widget'),
    );
  }
}

class MyHomePage extends StatefulWidget {
  MyHomePage({Key key, this.title}) : super(key: key);

  // This widget is the home page of your application. It is stateful, meaning
  // that it has a State object (defined below) that contains fields that affect
  // how it looks.

  
  final String title;

  @override
  _MyHomePageState createState() => _MyHomePageState();
}

class _MyHomePageState extends State<MyHomePage> {
  int _counter = 0;

  void _incrementCounter() {
    setState(() {
     
      _counter++;
    });
  }

  @override
  Widget build(BuildContext context) {
    
    return Scaffold(
      appBar: AppBar(
        
        title: Text(widget.title),
      ),
      body: Center(
        // Center is a layout widget. It takes a single child and positions it
        // in the middle of the parent.
        child: Column(
         
          children: <Widget>[
            RaisedButton(onPressed: (){  
            
              showDialog(context: context, builder:(context)=>
                     MaterialAlertDialog(

                      height: 250,

                      title: "Exit",

                      message: "Are you sure you want to quit",

                     onPresedPositiveButton: (){
                       //add what you want for navigate
                     },

                      onPresedNegativeButton: (){
                        Navigator.pop(context);
                      },

                   ),
               );
            }, 
            child: Text('MaterialAlertDialog',style: TextStyle(color:Colors.white)),
             color:Colors.redAccent
            ),

            RaisedButton(onPressed: (){  
            
              showDialog(context: context, builder:(context)=>
                     AnimatedMaterialAlertDialog(

                      imagePath: "lib/assets/images/steth.png",
              
                      title: "Exit",

                      message: "Are you sure you want to quit",

                     onPresedPositiveButton: (){
                       //add what you want for navigate
                     },

                      onPresedNegativeButton: (){
                        Navigator.pop(context);
                      },

                   ),
               );
            }, 
            child: Text('Animated MaterialAlertDialog',style: TextStyle(color:Colors.white)),
             color:Colors.redAccent
            )
          ],
        ),
      ),
    );
  }
}


lib/animatedmaterial_alertdialog.dart
import 'package:flutter/material.dart';


class AnimatedMaterialAlertDialog extends StatelessWidget {
  final double height;
  final String imagePath;
  final String title;
  final String message;
  final Function onPresedPositiveButton;
final Function onPresedNegativeButton;
   const AnimatedMaterialAlertDialog({
    Key key,
    this.height = 350.0,
    @required this.imagePath,
    @required this.message,
    @required this.onPresedPositiveButton,
    @required this.onPresedNegativeButton,
    @required this.title,
  }) : super(key: key);


  @override
  Widget build(BuildContext context){
   
    return Dialog(
      shape: RoundedRectangleBorder(
        borderRadius: BorderRadius.circular(16)
      ),
      elevation: 0,
      backgroundColor: Colors.transparent,
      child: _buildChild(context),
    );  
  }
  
  _buildChild(BuildContext context) => Container(
    height: 290,
    decoration: BoxDecoration(
      color: Colors.redAccent,
      shape: BoxShape.rectangle,
      borderRadius: BorderRadius.all(Radius.circular(18))
    ),
    child: Column(
      children: <Widget>[
        Container(
          child: Padding(
            padding: const EdgeInsets.all(8.0),
            child: Image.asset(imagePath,height: 120,width: 120,),
          ),
          width: double.infinity,
      decoration: BoxDecoration(
      color: Colors.white,
      shape: BoxShape.rectangle,
      borderRadius: BorderRadius.all(Radius.circular(18))
    ),
        ),
        SizedBox(height:20),
        Text(title,style: TextStyle(fontSize:20,color:Colors.white,fontWeight: FontWeight.bold),),
        SizedBox(height:15),
        Padding(
          padding: const EdgeInsets.only(right: 10,left: 10),
          child: Text(message,style: TextStyle(fontSize:15,color:Colors.white),textAlign:TextAlign.center,),
        ),
        SizedBox(height:20),
        Row(
          mainAxisSize: MainAxisSize.min,
          children: <Widget>[
            FlatButton(onPressed: (){
              onPresedNegativeButton();
              
            }, child: Text('No'),),
            RaisedButton(onPressed: (){
              onPresedPositiveButton();
            }, 
            child: Text('Yes',style: TextStyle(color:Colors.redAccent)),
            color:Colors.white),
          ],
        )
      ],
    ),
  );
}

  

lib/material_alertdialog.dart
library material_alertdialog;

import 'package:flutter/material.dart';


class MaterialAlertDialog extends StatelessWidget {
  final double height;
  final String imagePath;
  final String title;
  final String message;
  final Function onPresedPositiveButton;
final Function onPresedNegativeButton;
   const MaterialAlertDialog({
    Key key,
    this.height = 350.0,
    @required this.imagePath,
    @required this.message,
    @required this.onPresedPositiveButton,
    @required this.onPresedNegativeButton,
    @required this.title,
  }) : super(key: key);


  @override
  Widget build(BuildContext context){
   
    return Dialog(
      shape: RoundedRectangleBorder(
        borderRadius: BorderRadius.circular(16)
      ),
      elevation: 0,
      backgroundColor: Colors.transparent,
      child: _buildChild(context),
    );  
  }
  
  _buildChild(BuildContext context) => Container(
    height: 330,
    decoration: BoxDecoration(
      color: Colors.redAccent,
      shape: BoxShape.rectangle,
      borderRadius: BorderRadius.all(Radius.circular(18))
    ),
    child: Column(
      children: <Widget>[
        Container(
          child: Padding(
            padding: const EdgeInsets.all(8.0),
            child: Image.asset(imagePath,height: 120,width: 120,),
          ),
          width: double.infinity,
      decoration: BoxDecoration(
      color: Colors.white,
      shape: BoxShape.rectangle,
      borderRadius: BorderRadius.all(Radius.circular(18))
    ),
        ),
        SizedBox(height:30),
        Text(title,style: TextStyle(fontSize:20,color:Colors.white,fontWeight: FontWeight.bold),),
        SizedBox(height:15),
        Padding(
          padding: const EdgeInsets.only(right: 10,left: 10),
          child: Text(message,style: TextStyle(fontSize:15,color:Colors.white),textAlign:TextAlign.center,),
        ),
        SizedBox(height:34),
        Row(
          mainAxisSize: MainAxisSize.min,
          children: <Widget>[
            FlatButton(onPressed: (){
              onPresedNegativeButton();
              
            }, child: Text('No'),),
            RaisedButton(onPressed: (){
              onPresedPositiveButton();
            }, 
            child: Text('Yes',style: TextStyle(color:Colors.redAccent)),
            color:Colors.white),
          ],
        )
      ],
    ),
  );
}

  